tools/RepositoryManagement/DevHome.RepositoryManagement/Services/RepositoryEnhancerService.cs

```diff
@@ -0,0 +1,135 @@
+
+    private readonly Window _window;
+
+    private readonly ITelemetry _telemetry;
+
+    private readonly List<IExtensionWrapper> _repositorySourceControlProviders;
+
+    public RepositoryEnhancerService(
+        FileExplorerViewModel sourceControlRegistrar,
+        Window window,
+        IExtensionService extensionService)
+    {
+        _sourceControlRegistrar = sourceControlRegistrar;
+        _telemetry = TelemetryFactory.Get<ITelemetry>();
+        _window = window;
+        _repositorySourceControlProviders = extensionService.GetInstalledExtensionsAsync(ProviderType.LocalRepository).Result.ToList();
+    }
+
+    /// <summary>
+    /// associates a source control provider with a local repository.
+    /// </summary>
+    /// <param name="repositoryLocation">The full path to the repositories root.</param>
+    /// <returns>The guid of the class.  Otherwise Guid.Empty.</returns>
+    public async Task<Guid> MakeRepositoryEnhanced(string repositoryLocation)
+    {
+        _sourceControlRegistrar.AddRepositoryAlreadyOnMachine(repositoryLocation);
+        var sourceControlId = await AssignSourceControlToPath(repositoryLocation);
+        return sourceControlId;
+    }
+
+    public string GetLocalBranchName(string repositoryLocation)
+    {
+        try
+        {
+            using var repository = new LibGit2Sharp.Repository(repositoryLocation);
+            return repository.Head.FriendlyName;
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Error when getting branch name");
+        }
+
+        return string.Empty;
+    }
+
+    public string GetRepositoryUrl(string repositoryLocation)
+    {
+        try
+        {
+            using var repository = new LibGit2Sharp.Repository(repositoryLocation);
+            return repository.Network.Remotes.First().Url;
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Error when getting the repositoryUrl");
+        }
+
+        return string.Empty;
+    }
+
+    public IPropertySet GetProperties(string[] propertiesToReturn, string repositoryLocation)
+    {
+        var sourceControlProvider = new SourceControlProvider();
+        var provider = sourceControlProvider.GetProvider(repositoryLocation);
+
+        if (provider == null)
+        {
+            _log.Warning($"Path: {repositoryLocation} does not have an associated provider.");
+            return new PropertySet();
+        }
+
+        // This call does check the settings for file explorer and souce control integration.
+        return provider.GetProperties(propertiesToReturn, string.Empty);
+    }
+
+    private async Task<Guid> AssignSourceControlToPath(string maybeRepositoryPath)
+    {
+        if (string.IsNullOrEmpty(maybeRepositoryPath))
+        {
+            _log.Information("maybeRepositoryPath is null or empty.  Not assigning a source control provider");
+            return Guid.Empty;
+        }
+
+        if (!Directory.Exists(maybeRepositoryPath))
```

Comment:
Nit: I think `Directory.CreateDirectory()` succeeds and the return value reflects the existing directory if it already exists. Maybe just skip the `Exists` check and unconditionally call `CreateDirectory()`?

Reply:
You are correct.  I'm so used to "create" methods throwing if the object exists.  Changed.